lib/ingestors/ingestor_event_rest.rb

```diff
@@ -401,6 +417,226 @@ def process_elixir(url)
       end
     end
 
+    def process_vu(url)
+      # execute REST request
+      results = get_JSON_response url
+      data = results.to_h['results']
+
+      # extract materials from results
+      unless data.nil? or data.size < 1
+        data.each do |item|
+          begin
+            # create new event
+            event = Event.new
+
+            # extract event details from
+            attr = item
+            event.title = attr['title']
+            event.url = attr['url'].strip unless attr['url'].nil?
+            event.organizer = attr['org']
+            event.description = convert_description attr['description']
+            event.start = attr['startdt']
+            event.end = attr['enddt']
+            event.venue = attr['location']
+            event.city = 'Amsterdam'
+            event.country = 'The Netherlands'
+            event.source = 'VU'
+            event.online = attr['online_event']
+            event.contact = attr['orgurl']
+            event.timezone = 'Amsterdam'
+
+            # array fields
+            event.keywords = []
+            attr['categories_arr'].each { |category| event.keywords << category['name'] } unless attr['categories'].nil?
+
+            event.event_types = []
+            unless attr['event_types'].nil?
+              attr['event_types'].each do |key|
+                value = convert_event_types(key)
+                event.event_types << value unless value.nil?
+              end
+            end
+
+            event.target_audience = []
+            attr['audiences'].each { |audience| event.keywords << audience.name } unless attr['audience'].nil?
+
+            event.host_institutions = []
+            attr['host-institutions'].each { |host| event.host_institutions << host } unless attr['host-institutions'].nil?
+
+            # dictionary fields
+            event.eligibility = []
+            unless attr['eligibility'].nil?
+              attr['eligibility'].each do |key|
+                value = convert_eligibility(key)
+                event.eligibility << value unless value.nil?
+              end
+            end
+
+            # add event to events array
+            add_event(event)
+            @ingested += 1
+          rescue Exception => e
+           @messages << "Extract event fields failed with: #{e.message}"
+           raise e if true
+          end
+        end
+      end
+    end
+
+    def process_surf(url)
+      Hash.from_xml(Nokogiri::XML(URI.open(url)).to_s)['sitemapindex']['sitemap'].each do |page|
```

Comment:
probably you can avoid `Hash.from_xml` and use Nokogiri methods to select what you want to use.